src/Process/JobStateChangelogProcessor.php

```diff
@@ -4,28 +4,67 @@
 namespace Neighborhoods\Kojo\Process;
 
 use Neighborhoods\Kojo\Semaphore;
+use Neighborhoods\Kojo\JobStateChange;
 
 class JobStateChangelogProcessor extends Forked implements JobStateChangelogProcessorInterface
 {
     use Semaphore\Resource\Factory\AwareTrait;
+    use JobStateChange\Repository\AwareTrait;
 
     public const TYPE_CODE = 'job_state_changelog_processor';
 
+    protected $batchSize;
+
     protected function _run(): Forked
     {
         $this->_getLogger()->debug('JobStateChangelogProcessor has been instantiated');
+        $semaphoreResource = $this->_getSemaphoreResource(self::TYPE_CODE);
 
-        if ($this->_getSemaphoreResource('job_state_changelog_processor')->testAndSetLock()) {
+        if ($semaphoreResource->testAndSetLock()) {
             $this->_getLogger()->debug('JobStateChangelogProcessor has acquired mutex');
-
-            // TODO: replace with business logic
-            while (true) {
-                sleep(1);
-            }
+            $this->businessLogic();
+            $semaphoreResource->releaseLock();
```

Comment:
Not totally necessary since the lock will be released once this process exits, but I like the explicitness